Shared/ViewModels/SettingsViewModel.swift

```diff
@@ -44,4 +55,26 @@ final class SettingsViewModel: ObservableObject {
 		}.sorted(by: { $0.name < $1.name })
 		self.langs.insert(.auto, at: 0)
 	}
+
+	func sendQuickConnect() {
+		QuickConnectAPI.authorize(code: self.quickConnectCode)
+			.sink(receiveCompletion: { completion in
+				switch completion {
+				case .failure:
+					LogManager.log.debug("Invalid Quick Connect code entered")
+					self.validQuickConnect = false
+				default:
+					self.handleAPIRequestError(displayMessage: "Error", completion: completion)
+				}
+			}, receiveValue: { value in
+				if !value {
+					LogManager.log.debug("Invalid Quick Connect code entered")
+					self.validQuickConnect = false
+				} else {
+					LogManager.log.debug("Valid Quick connect code entered")
+					self.router?.dismissCoordinator()
```

Comment:
I don't think we should dismiss the coordinator, so the router reference isn't necessary as well.